### PR TITLE
Stop using deprecated `GdkPixbuf.from_xpm_data`

### PR DESCRIPTION
**Pull request by tleedjarv**
Re #1033 #1034

Since the XPM format is so simple, the function to parse (subset of) XPM used for the icons is trivial. This approach does not require external resources (like image files or loader plugins) nor embedding binary-format icons in the source.

Working but not heavily tested.

### CODE REVIEW

File: src/dune [hunk 1]
```diff
@@ -1,7 +1,7 @@
 (library
  (name unison_lib)
  (wrapped false)
- (modules :standard \ linktext linkgtk3 uigtk3 uimacbridge test)
+ (modules :standard \ linktext linkgtk3 uigtk3 pixmaps uimacbridge test)
```
Comment (gdt) [inline src/dune:4]: commit message says "fix", but I think this is "add module that the previous commit started using".  Always watching out for ways future self might misread....

**Comment by gdt** [comment]
I would really like to see someone review and test.    We just had a release, and I am going to try not to merge anything for a bit until the release and the gdk-pixbuf-is-unstable kerfluffle settle out a bit.

File: src/dune (hunk 1 above)
Comment (tleedjarv) [reply src/dune:4]: I considered it a fix as the list of modules was incorrect even before the pixbuf change. It's just that it didn't produce an actual error before the pixbuf change, but it wasn't correct. The pixbuf change made the Pixmaps module depend on lablgtk.

**Comment by ithod** [comment]
On openSUSE Tumbleweed I get the following error:
`Uncaught exception (please report a bug): Gpointer.Null
Raised by primitive operation at Uigtk3.Private.createToplevelWindow.buildPixmap in file "/home/abuild/rpmbuild/BUILD/unison-2.53.5/src/uigtk3.ml" (inlined), line 3296, characters 4-29
Called from Uigtk3.Private.createToplevelWindow in file "/home/abuild/rpmbuild/BUILD/unison-2.53.5/src/uigtk3.ml", line 3300, characters 17-44
Called from Uigtk3.Private.start in file "/home/abuild/rpmbuild/BUILD/unison-2.53.5/src/uigtk3.ml", line 4489, characters 19-42`

https://build.opensuse.org/project/show/Archiving:unison

**Comment by gdt** [comment]
Are you testing the branch in this PR, or an artifact built by CI from this branch?  It is known that a release build of unison with a buggy gdk-pixbuf (as contained in openSUSE) does not work.

**Comment by tleedjarv** [comment]
@ithod I'm pretty certain that https://build.opensuse.org/project/show/Archiving:unison does not contain the patch from this PR. This makes your comment a duplicate of #1033.

**Comment by ithod** [comment]
> I would really like to see someone review and test. We just had a release, and I am going to try not to merge anything for a bit until the release and the gdk-pixbuf-is-unstable kerfluffle settle out a bit.

Sorry for my unclear message, indeed the error occurs if the patch is not applied. I wanted to make clear that this patch solves the error given above. When I apply the patch (the 1st, not sure what `src/dune` is), it works for me
https://build.opensuse.org/package/show/home:ithod:updated/unison

**Comment by tleedjarv** [comment]
Thank you @ithod.

> the patch (the 1st, not sure what `src/dune` is)

The 1st is the real patch, yes. The other one is only needed when building with `dune` (and not `make`).

**Comment by gdt** [comment]
We've had one test report, and enough time that anybody that wanted to test has had a chance.

**Pull request merged**